### ISSUE TITLE
Revisit PWA support

### ISSUE THREAD

**Issue by jbaicoianu** [jbaicoianu/janusweb#225]
We implement PWA support via an app manifest a couple years ago, but disabled it since there wasn't really any support for it on the platforms where we felt it really mattered. Now that Oculus Browser supports PWAs as a way to install webapps to the Quest home screen, it's probably time to revisit the feature

**Comment by jbaicoianu** [comment]
Another thing to revisit is the service worker. We can use this to do things like make each browser tab aware of what other tabs are open, and do notifications cross-tab, and for offline users

**Comment by jbaicoianu** [comment]
Looks like I misunderstood the announcement, and there's currently no way to install directly from the website on Quest like you can from Desktop and Mobile - Facebook seems to be planning to gate this behind AppLab and the App Store.  It's currently possible to build an APK out of a PWA and sideload it though - we can probably set up a build step to do this, and can evaluate whether it makes sense to put it on AppLab at such time as they make that an option

https://timmykokke.com/blog/2021-11-05-webxr-pwa-apk/

**Comment by jbaicoianu** [comment]
Ugh. No Linux build tools, proprietary manifest extensions. These are not Progressive Web Apps, they're Oculus Web Apps. :angry:

**Comment by jbaicoianu** [comment]
I've successfully built and installed a PWA for running JanusWeb on the Oculus Quest. Still miffed that I have to do this from windows or OSX - I made some attempts to run it via WINE, but ran into some issues which I couldn't resolve. At least I know the project is set up properly for this now, and maybe one day Facebook will release Linux tools so more projects can integrate this into their standard build procedure.
